### PR TITLE
Coerce nested properties

### PR DESCRIPTION
**Pull request by luisfavila**
### Description

Loopback doesn't coerce nested properties. This intends to fix it.
There was a previous pull request (#1515) which was eventually forgotten. @bajtos asked me to open a new one, but the juggler code has changed since then so this pull request was rewritten to accommodate the changes.

#### Related issues
- in sequence of #1515 
- connect to strongloop/loopback#517
- connect to #1514 
- connect to #1169

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by bajtos** [comment]
@raymondfeng @jannyHou you are more familiar with this part of our codebase, could you PTAL?

File: test/nested-properties-test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+              'nested.happenedOn': {
+                'gt': new Date('2017-10-20T06:07:31.805Z'),
+              },
+            },
+          });
+        })
+        .then(found => {
+          should.exist(found);
+          found.should.not.be.empty();
+          found[0].should.be.an.instanceOf(Model);
```
[the remark this answers is not in the document]
Comment (luisfavila) [reply test/nested-properties-test.js]: Done!

**Comment by luisfavila** [comment]
Resolved conflicts with master

**Comment by radarsu** [comment]
Any update?

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
Sorry for the late reply, I will review it today.

**Comment by jannyHou** [comment]
@b-admike I noticed that you are also doing some refactor for the nested property coercion, could you also review this PR(hopefully there is no overlapping)? Thanks!

**Comment by b-admike** [comment]
@jannyHou Sorry for the late reply. Will take a look this week.

**Comment by melote** [comment]
Please check this and merge it, we really need this feature!

**Comment by webocs** [comment]
Any news on this?? This PR seems finished and tested... Why is it not merged?

**Comment by bajtos** [comment]
It looks like this PR sort of fell through the cracks and was forgotten.

From my point of view, it's important to find a solution that will work for all connectors supported by the LoopBack core team. Most notably, the MongoDB connector already implements coercion of nested properties in queries.

**Pull request closed without merging**